api/app/signals/apps/email_integrations/rules/abstract.py

```diff
@@ -64,3 +66,19 @@ def _validate_status(self, status):
         overwrite this function in the defined Rule to add the additional status validation checks
         """
         return False
+
+    def _validate_allows_contact(self, signal):
+        """
+        Validate if the user want to be contacted and if allows_contact on feedback is False t
+        o never send ANY emails to the user
+
+        If the feature flag is False we return True to still send emails to the user.
+        """
+        if not settings.FEATURE_FLAGS.get('REPORTER_MAIL_DISABLE_CONTACT_FEEDBACK_ALLOWS_CONTACT', True):
+            return True
+
+        try:
+            return signal.feedback.last().allows_contact
```

Comment:
This should only check for feedback that has been submitted. No need to check for feedback objects that are not submitted. So just add a filter to check if the `submitted_at` isset (and order by it because you need the latest one).